04-onion-routing.md

```diff
@@ -76,81 +70,59 @@ The use of a fixed nonce is safe since the keys are never reused.
 
 ## Packet Structure
 
-The packet consists of 2 parts:
+The packet consists of 4 parts:
  
- - The fixed size _header_ containing meta information about the
-   packet and the routing information necessary to forward the
-   message.
- - A fixed size _per-hop payload_ containing information for each hop
+ - A `version` byte
+ - A 33-byte compressed `secp256k1` `public key`, used during the shared secret generation
+ - A 1060-byte `hops_data` consisting of twenty fixed size packets containing information for each hop
    as they forward the message.
+ - A 20-byte `HMAC` used to verify the packet's integrity.
 
 The overall structure of the packet is depicted below. The network format of the packet consists of the individual parts being serialized into one continguous byte-stream and then transferred to the recipient of the packet. Due to the fixed size of the packet it does not need to be prefixed by its length when transferred over a connection.
 
 ~~~~
-+--------+-----------------+
-| header | per-hop payload |
-+--------+-----------------+
++------------------+-----------------------+------------------------+-----------------+
+| version (1 byte) | public key (33 bytes) | hops_data (20x53 bytes) | HMAC (20 bytes) |
```

Comment:
In this proposal, since the 20-byte `next_hop` field has been eliminated, the MAC's no longer need to be 20 bytes. If we move forward with this, we can instead modify the packet-wide MAC to be 32-bytes.